lib/grape-swagger.rb

```diff
@@ -231,7 +231,9 @@ def strip_heredoc(string)
             end
 
             def parse_base_path(base_path, request)
-              (base_path.is_a?(Proc) ? base_path.call(request) : base_path) || request.base_path
+              url_scheme = request.env["rack.url_scheme"]
+              (base_path.is_a?(Proc) ? base_path.call(request) : base_path) ||
+                "#{url_scheme}://#{request.env['HTTP_HOST']}#{url_scheme == 'https' ? ':80' : ''}"
```

Comment:
Does this mean set the port to `80` if the scheme is `https`?  That seems counter intuitive at first glance.  What's the background?